src/environments/array.js

```diff
@@ -48,11 +48,14 @@ function getHLines(parser: Parser): boolean[] {
 
 const validateAmsEnvironmentContext = context => {
     const settings = context.parser.settings;
+    const input = context.parser.gullet.lexer.input.trim();
     if (!settings.displayMode) {
-        throw new ParseError(`{${context.envName}} cannot be used inline.`);
-    } else if (settings.strict && !settings.topEnv) {
-        settings.reportNonstrict("textEnv",
-            `{${context.envName}} called from math mode.`);
+        throw new ParseError(`{${context.envName}} can be used only in` +
+            ` display mode.`);
+    } else if (!/^\\begin{/.test(input)) {
```

Comment:
There may be edge cases, like the environment used via macro, e.g., `\def\foo#1{\begin{equation}#1\end{equation}}`

Reply:
Thank you for catching that. I am closing this PR and will attack the problem in a new PR.